engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnEventSubscriptionBehavior.java

```diff
@@ -251,14 +242,40 @@ private void publishTriggeredEvent(
         .findFirst()
         .ifPresent(
             deferredRecord ->
-                publishActivatingEvent(deferredRecord.getKey(), deferredRecord.getValue()));
+                publishActivatingEvent(
+                    context, deferredRecord.getKey(), deferredRecord.getValue()));
   }
 
+  // TODO (saig0): replace by findEventTrigger + activateTriggeredEvent (#6202)
   private void publishActivatingEvent(
-      final long elementInstanceKey, final ProcessInstanceRecord eventRecord) {
+      final BpmnElementContext context,
+      final long elementInstanceKey,
+      final ProcessInstanceRecord eventRecord) {
+
+    if (MigratedStreamProcessors.isMigrated(eventRecord.getBpmnElementType())) {
+      if (context.isInReprocessingMode()) {
+        // don't wring events/commands on reprocessing
+        return;
+      }
```

Comment:
This check is necessary to avoid that a non migrated element processor (e.g. service task) apply the `activating` event for the boundary event twice.